packages/sanity/src/core/form/inputs/arrays/ArrayOfPrimitivesInput/ArrayOfPrimitivesInput.tsx

```diff
@@ -192,27 +196,33 @@ export class ArrayOfPrimitivesInput extends React.PureComponent<
                 >
                   {membersWithSortIds.map(({member, id}, index) => {
                     return (
-                      <Item
+                      <ChangeIndicator
                         key={member.key}
-                        id={id}
-                        sortable={isSortable}
-                        disableTransition={this.state.disableTransition}
+                        path={member.kind === 'item' ? member.item.path : path}
```

Comment:
Needed to avoid typing issues. Decided to use the same logic as what we use in line 210

Reply:
I don't think this will render correctly for error items - think it would be better if you could move the <Item .. element to inside the {member.kind === 'item'} check below and wrap it there instead?